### ISSUE TITLE
Exception: input tree needs to have a chop

### ISSUE THREAD

**Issue by afshawnlotfi** [gregstarr/pychop3d#34]
Hi so one of the main reasons I was interested in this project is I wanted to 3D print an airplane wing STL file and then connect it all together. For some reason no matter how much I change the config it keeps giving me this exception: model.stl is an airplane wing design.

```
Using config: examples/model_config.yml
Using mesh: examples/model.stl
2020-08-26 22:00:39,178  pychop3d.search  [INFO]  Starting beam search with an instance of <class 'trimesh.base.Trimesh'>
2020-08-26 22:00:39,180  __main__  [INFO]  Best BSP-tree found in 0.02323007583618164 seconds
2020-08-26 22:00:39,185  __main__  [INFO]  finding best connector arrangement
1
Traceback (most recent call last):
  File "main.py", line 119, in <module>
    run(starter)
  File "main.py", line 61, in run
    connector_placer = connector.ConnectorPlacer(tree)
  File "/workspaces/pychop3d/pychop3d/connector.py", line 25, in __init__
    raise Exception("input tree needs to have a chop")
Exception: input tree needs to have a chop
```
Here is my config:

```yml
beam_width: 2 
connector_diameter: 2 
connector_spacing: 10
connector_tolerance: 10
mesh: "examples/model.stl" 
part_separation: true 
printer_extents: 
- 200.0
- 200.0
- 200.0
directory: "output" 
name: "model"
scale_factor: 2
```

My stl file is attached:
[https://drive.google.com/file/d/1JtOYYp7UZ_EfD7JNiLHOih5WrDAChaKf/view?usp=sharing](https://drive.google.com/file/d/1JtOYYp7UZ_EfD7JNiLHOih5WrDAChaKf/view?usp=sharing)

I don't know what exactly the problem is but this worked great for the bunny stl. Does this only work with straight edges?

**Comment by gregstarr** [comment]
I believe the problem here is that the STL units are not in mm. You can use the `scale_factor` config setting to make the appropriate conversion, e.g. if the units are in cm, you can set `scale_factor` to 10. Keep the `connector_diameter` at 5 or 10 (mm) because the wing looks thin. I would say increase the `beam_width` to 3

**Comment by gregstarr** [comment]
But another issue is that the wing seems extremely thin, try increasing its thickness a bit

**Comment by afshawnlotfi** [comment]
Yeah that's what I was thinking as well

**Comment by gregstarr** [comment]
I tried running this example myself. The original problem you were having was that the wing was already small enough to fit in the configured printer, the default is 200x200x200mm. I assumed that the STL units were in inches and configured the scale factor to 25.4. This was big enough to require at least 1 partition, but it failed to chop because all of the cross-sections were too small to comfortably fit a connector on. Just for the purpose of demonstration, I doubled the size of the wing again and changed the connector size to an impractical 2mm. This worked for me both on my windows computer and in the docker container. Try running the new example by cd'ing to the pychop3d directory and calling:
```
python main.py -c examples/wing_config.yml
```

If you are satisfied with this resolution, let me know and I can close this issue

**Comment by afshawnlotfi** [comment]
Oh my gosh! You are awesome. I tried forever to fix this and realized that the library I am using [Machup](https://github.com/usuaero/MachUp_Py) is in inches by default. Thanks so much. Before I close I just wanted to ask if you changed the STL at all because I noticed it got converted from ASCII to Binary format and if so how did you change it for future reference.

**Comment by gregstarr** [comment]
Yes I forgot to mention that I reduced the number of triangles in the original STL to make it run faster. I did this by opening Blender, importing the STL and applying the "decimate" modifier with a factor of .1. I believe this should reduce the number of triangles to 1/10th of the original. The reduced STL is in the examples folder. I think the original should still work though. As for it being in binary, I think that is just the default when saving with Blender.
